tests/sentry/snuba/test_discover.py

```diff
@@ -567,6 +579,99 @@ def test_params_forward(self, mock_query):
             referrer=None,
         )
 
+    @patch("sentry.snuba.discover.raw_query")
+    def test_aggregate_conditions(self, mock_query):
+        mock_query.return_value = {
+            "meta": [{"name": "transaction"}, {"name": "duration"}],
+            "data": [{"transaction": "api.do_things", "duration": 200}],
+        }
+        start_time = before_now(minutes=10)
+        end_time = before_now(seconds=1)
+        discover.query(
+            selected_columns=["transaction", "avg(transaction.duration)"],
+            query="http.method:GET avg(transaction.duration):>5",
+            params={"project_id": [self.project.id], "start": start_time, "end": end_time},
+        )
+        mock_query.assert_called_with(
+            selected_columns=["transaction"],
+            conditions=[["http_method", "=", "GET"]],
+            filter_keys={"project_id": [self.project.id]},
+            groupby=["transaction"],
+            dataset=Dataset.Discover,
+            aggregations=[["avg", "duration", "avg_transaction_duration"]],
+            having=[["avg_transaction_duration", ">", 5]],
+            end=end_time,
+            start=start_time,
+            orderby=None,
+            limit=50,
+            offset=None,
+            referrer=None,
+        )
+
+    @patch("sentry.snuba.discover.raw_query")
+    def test_alias_aggregate_conditions(self, mock_query):
+        mock_query.return_value = {
+            "meta": [{"name": "transaction"}, {"name": "duration"}],
+            "data": [{"transaction": "api.do_things", "duration": 200}],
+        }
+        start_time = before_now(minutes=10)
+        end_time = before_now(seconds=1)
+        with pytest.raises(InvalidSearchQuery):
+            discover.query(
+                selected_columns=["transaction", "p95"],
+                query="http.method:GET p95:>5",
```

Comment:
In this situation are we not able to use `p95 > 5` in the having clause directly? Clickhouse supports using alias names in `HAVING` conditions.

Reply:
I thought we had to map p95 to `quantile(0.95)(duration)` before we sent it to snuba/clickhouse.

Reply:
We do that as part of the select clause (aggregrations) once it is there, we can reuse the `p95` alias in the having conditions.

Reply:
OK I'll test that out.

Reply:
Got this working!